### PR TITLE
New trait which completely forbids using the overmap

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Content "New trait which completely forbids using the overmap"

#### Purpose of change
More content. New challenge-type, game-changing trait.

#### Describe the solution
- Added a trait that completely forbids using the overmap. This means "completely": no overmap, no looking at the sky, no map through zones manager, and no faction camp missions that require using the overmap. Even no more debug mode editor.
- Also forbid using maps as they are useless without overmap.
- The only tools of navigating left are small overmap widget in sidebar and minimap.
- Trait costs -2 points. Cost could be increased or decreased if needed.
- Trait is mutually exclusive with Scout and Topographagnosia traits as they are useless without overmap.

#### Describe alternatives you've considered
- Replace effect of Topographagnosia with this new effect instead of creating a new trait.
- Allow the only way of using overmap - through faction camp missions UI.

#### Testing
Got new trait. Tried opening map through `M`, through action menu, through zones manager, through faction camp missions UI.

#### Additional context
None.

### CODE REVIEW

**Comment by Fris0uman** [comment]
Wouldn't this make more sense as a world setting?

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -8052,5 +8052,16 @@
     "purifiable": false,
     "starting_trait": true,
     "flags": [ "NYCTOPHOBIA" ]
+  },
+  {
+    "type": "mutation",
+    "id": "NO_OVERMAP",
+    "name": { "str": "Visuospatial dysgnosia" },
+    "points": -2,
+    "description": "Due to serious brain damage, you have significant troubles with determining your whereabouts.  You can barely navigate your immediate surroundings.  You don't understand maps, and you can't use overmap.",
```
Comment (NetSysFire) [inline data/json/mutations/mutations.json:8061]:
```suggestion
    "description": "Due to serious brain damage, you have significant troubles with determining your whereabouts.  You can barely navigate your immediate surroundings.  You don't understand maps, and you can't use the overmap.",
```

**Review by NetSysFire — COMMENTED** [review]
The overmap is more or less a critical feature of the game. Even if this is an optional challenge, I am not entirely sure if it is a good idea. Mostly because the overmap is a massive quality of life feature. I would increase the point cost in any case, maybe even to -5 or even lower. Imagine finding fungaloids and while you search for the source they spread in the meantime because they cover an immense amount of land, same for ants. You could also run into hordes and portal/mass grave map extras, causing extra frustration when flaming eyes are involved.

Another thing that comes to mind is that the overmap widget will probably still work, allowing partial circumvention by using the sidebar and autonote marking map extras.

**Comment by Zireael07** [comment]
Seconding NetSysFire that the cost should be higher.

Also: why forbid looking at the sky? The sky/weather has nothing to do with using a map, which is what the overmap presumably represents...

**Comment by Zireael07** [comment]
@Night-Pryanik but that's just an implementation detail, not related to what the character in-game is doing (which is what the traits imho represent) - IMHO looking at the sky should work but the normal overmap should not. 
Yes I know it will be slightly more involved than just making the overmap button not do anything, but still would make more sense

**Comment by Night-Pryanik** [comment]
> The overmap is more or less a critical feature of the game. Even if this is an optional challenge, I am not entirely sure if it is a good idea. Mostly because the overmap is a massive quality of life feature.

You want the challenge, you take this trait. You don't want to lose the qol feature, you don't take this trait. Easy.

> Another thing that comes to mind is that the overmap widget will probably still work, allowing partial circumvention by using the sidebar and autonote marking map extras.

Have you read the OP? I wrote that overmap widget and minimap are still working. I didn't remove them by design.

**Comment by Night-Pryanik** [comment]
> IMHO looking at the sky should work but the normal overmap should not.

How do you imagine it? Looking at the sky literally opens the overmap, on the 10th z-level. You can't prevent player from looking to the lower levels, and it just breaks the challenge.

**Comment by Night-Pryanik** [comment]
> Blocking the overmap completely would not only prevent using maps, it also restricts your vision range in a way. I.e. your overmap vision range, which is larger than what you really "see".

This is intended. As said in trait's description, you can barely navigate your immediate surroundings. You can't remember and navigate your remote surroundings at all.

> No overmap memory. You only see what you currently see / is in vision range

This might be a good idea, but it's definitely more difficult to implement. I'm not sure I want to delve that deep into the overmap displaying code.

> the overmap widget's radius is not fixed. So one could use it to see quite a bit of the map if used in sufficient size.

That's ok. The goal is to limit player's ability to see vast distances, not immediate surroundings.

**Comment by Zireael07** [comment]
> That's ok. The goal is to limit player's ability to see vast distances, not immediate surroundings.

In that case, I think we already have a trait that limits the distances? Trophosomething?

> How do you imagine it? Looking at the sky literally opens the overmap, on the 10th z-level. You can't prevent player from looking to the lower levels,

Literally what I wrote, if has trait then disable changing Z-levels in overmap

**Comment by Night-Pryanik** [comment]
> I think we already have a trait that limits the distances? Trophosomething?

That's not the only goal. Overmap allows overmap memory, searching, auto-notes and many other features which Topographagnosia didn't cope with.

**Comment by AtomicFox556** [comment]
Not sure if this is a good idea; the player could just draw map manually on paper or using a text editor, in which case it's adding more tedium than challenge.

**Comment by SariusSkelrets** [comment]
If they do draw a map, it's because they decided to do it to mitigate the challenge they decided to took.
Neither the challenge nor the map drawing are forced, they can not do it but they did anyway. It's not like if the game forced you to do it.

If the player is willing to put that level of effort to mitigate the challenge, then let them do it

**Comment by Inglonias** [comment]
The only part of this I don't like is limiting the debug menu. The debug menu is for, well... debugging. You shouldn't limit player tools for debugging. If a player wants to use the debug menu to cheat their way around this challenge, that's on them.

**Comment by Night-Pryanik** [comment]
> The only part of this I don't like is limiting the debug menu. The debug menu is for, well... debugging. You shouldn't limit player tools for debugging. If a player wants to use the debug menu to cheat their way around this challenge, that's on them.

Well, if player _really_ wants to use the map-editing part of debug menu, he can temporarily remove the trait and do all he wants. It's a workaround, I know, it works.

All in one, the trait limiting debug menu, look at the sky and similar stuff is purely technical stuff (and a bit of laziness), not a design decision. It's much more easier to place one check in one place than add a ton of checks for tons of edge cases in tons of places throughout the code.

> You can make that sidebar overmap widget as big as you want. A dedicated min-maxer could get a pseudo-overmap by making a big enough sidebar and overmap widget. 

Even if you make the widget really big, it won't be the replacement for a proper overmap. You can't search the map with a widget, you can't scroll the map with a widget - it still shows only your immediate surroundings, you can't place notes on the widget, and so on. So I personally don't consider this widget 
as a "threat" to the challenge.

**Comment by Night-Pryanik** [comment]
Regarding the idea of no overmap memory: no need to draw the map on paper or using text editor. Notes to the rescue! Player enters some area, player opens map, player makes notes on interesting locations. Even if he later leaves this area and it becomes "unknown" once again, the notes are still here. And I don't think that removing ability to use notes (or erasing notes when player leaves the "known" area) is a good idea.

**Comment by mlange-42** [comment]
> Regarding the idea of no overmap memory: no need to draw the map on paper or using text editor. Notes to the rescue! Player enters some area, player opens map, player makes notes on interesting locations. Even if he later leaves this area and it becomes "unknown" once again, the notes are still here. And I don't think that removing ability to use notes (or erasing notes when player leaves the "known" area) is a good idea.

Allowing (auto)-notes, and deleting them when getting out of sight, was my conclusion when thinking in the no-memory-direction.

Maybe I should give another trait a try, let's say "Cartographobia". Could also have a funny result when trying to look at a (paper) map. "In a burst of fear sweat, you drop the map on the floor."

> Would it be possible to alter that widget to limit how much it draws?

Would probably be useful to limit it to the current overmap view range for Cartographobia.

**Comment by I-am-Erk** [comment]
Hmm. My issue with this is it feels very 'meta'. You could get the same effect by unbinding the key to open the overmap, it's not really something to do with your player. I think I'd prefer to see a Conduct or Achievement for it if you want it as a challenge, not a trait. Plus as many have pointed out, it disables a bunch of features secondarily that are reliant on the overmap, such as autodrive.

**Comment by Night-Pryanik** [comment]
> You could get the same effect by unbinding the key to open the overmap
> it disables a bunch of features secondarily that are reliant on the overmap, such as autodrive.

Unbinding key for opening overmap disables most overmap-reliant features, autodrive included, as well.

> I think I'd prefer to see a Conduct or Achievement for it if you want it as a challenge, not a trait.

"Don't open an overmap not a single time for the whole run"? It's not nearly as "serious" as being physically impossible to open it (not counting using debug mode).  Plus, it's REALLY easy to press `m` just by muscle memory, without using brain at all. And what next? Conduct failed. Very interesting (no).

> it feels very 'meta'.

Game provides a challenge (and a method to control it). So... Meta or no, who cares, if this is supposed to be interesting/fun?

**Comment by I-am-Erk** [comment]
To the best of my knowledge, there are no other traits in the game that represent some meta challenge, rather than an actual trait of your character. I don't think it's a great direction to go, personally.

My point is that you can achieve this effect without adding a new trait that doesn't fit with the rest, by adding a conduct and unbinding the key. It is functionally identical.

**Comment by anothersimulacrum** [comment]
> there are no other traits in the game that represent some meta challenge, rather than an actual trait of your character.

Wayfarer does as well.

**Comment by I-am-Erk** [comment]
Not really, PTSD can make it functionally impossible to do certain things. Wayfarer is still a function ban, not a UI ban. This is more akin to a trait that only shows you the first two letter of items in your inventory or something - it doesn't really reflect any kind of actual nature of the character. There's no unified trait that explains why you can't use your map, and also can't look at the weather, and also can't make long trips easily.

That said, IMO wayfarer should just have a strong disincentive to drive, but if that disincentive is something like "uncontrollable panic attack the moment you sit in the driver's seat" the issue is more an engine limitation, not a matter of what you're trying to simulate.

**Comment by Night-Pryanik** [comment]
>To the best of my knowledge, there are no other traits in the game that represent some meta challenge, rather than an actual trait of your character.

I'm not sure what did you mean by "meta challenge". And "meta" term in general is unclear to me as well.

>you can achieve this effect without adding a new trait that doesn't fit with the rest, by adding a conduct and unbinding the key. It is functionally identical.

Wayfarer: add a conduct and unbind "interact with vehicle" key.
Illiterate: add a conduct and unbind "read" key.
(not yet added) Brawler: add a conduct and unbind "fire" key.

I might be exaggerating a bit, but you get the point.

As far as I can tell, many players (myself included) don't give a shit about conducts and achievements, and even disable them altogether. Sure, they can unbind a key even with disabled conducts and achievements to create their own challenge. But player-enforced challenge is something very different from game-enforced challenge. 

I personally don't like saying anything like "Hey player, here's a new challenge we invented. But we won't do literally anything to make you follow this challenge, you yourself have to force yourself not to press a button. Cool innit?" instead of enforcing this challenge strictly through the game means.

**Comment by I-am-Erk** [comment]
> Wayfarer: add a conduct and unbind "interact with vehicle" key.

That's actually a great example of why I don't like this. I would have been staunchly against wayfarer if it prevented any interaction with vehicles, since that also blocks things like turning on a generator that have nothing to do with the wayfarer trait. Your proposed trait is like that: it doesn't have a coherent story to it, it just unbinds a command and everything that is downstream from it.

**Comment by kevingranade** [comment]
This is a no. As has been pointed out, if you want to challenge yourself just unbind the key.
And that's a no on trait, option, challenge, etc, the concept is the problem not the implementation.

**Comment by Inglonias** [comment]
In an effort to throw Night a bone, would "illiterate but for map items" have a chance to get through when this did not?

**Comment by esotericist** [comment]
> I'm not sure what did you mean by "meta challenge". And "meta" term in general is unclear to me as well.

'meta' generally refers to things *about* a thing without being *part* of the thing. that's a very brief version, though; there are better, more comprehensive definitions available on the internet.

the fundamental issue here i think is that you are targeting the human at the keyboard playing the game, rather than targeting the survivor the player is in charge of. that's what makes it 'meta', neither the goal or methodology are aimed at providing restrictions to the survivor.

**Comment by Night-Pryanik** [comment]
In that case both Scout and Topo have the same issue: since they increase/decrease overmap sight radius, they both target player, not the player's character.

**Pull request closed without merging**